platform-dom/src/main/java/com/softicar/platform/dom/elements/time/DomTimeInput.java

```diff
@@ -4,88 +4,98 @@
 import com.softicar.platform.common.core.interfaces.ITestMarker;
 import com.softicar.platform.common.date.IllegalTimeSpecificationException;
 import com.softicar.platform.common.date.Time;
+import com.softicar.platform.common.string.Tokenizer;
 import com.softicar.platform.dom.DomCssClasses;
 import com.softicar.platform.dom.DomI18n;
 import com.softicar.platform.dom.DomTestMarker;
-import com.softicar.platform.dom.elements.input.DomIntegerInput;
-import com.softicar.platform.dom.elements.label.DomPreformattedLabel;
 import com.softicar.platform.dom.input.AbstractDomValueInputDiv;
+import com.softicar.platform.dom.input.DomTextInput;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
 
 public class DomTimeInput extends AbstractDomValueInputDiv<Time> {
 
-	private final DomIntegerInput hourInput;
-	private final DomIntegerInput minuteInput;
-	private final DomIntegerInput secondInput;
+	private final DomTextInput timeInput;
 
 	public DomTimeInput() {
 
-		this.hourInput = createInput(DomI18n.HOURS, DomTestMarker.TIME_INPUT_HOURS_INPUT);
-		this.minuteInput = createInput(DomI18n.MINUTES, DomTestMarker.TIME_INPUT_MINUTES_INPUT);
-		this.secondInput = createInput(DomI18n.SECONDS, DomTestMarker.TIME_INPUT_SECONDS_INPUT);
-
+		this.timeInput = createInput(DomI18n.TIME, DomTestMarker.TIME_INPUT);
 		addCssClass(DomCssClasses.DOM_TIME_INPUT);
-		appendChildren(hourInput, new DomPreformattedLabel(":"), minuteInput, new DomPreformattedLabel(":"), secondInput);
+		appendChildren(timeInput);
+	}
+
+	public void setPlaceholder(IDisplayString placeholder) {
+
+		timeInput.setPlaceholder(placeholder);
 	}
 
 	@Override
 	protected void doSetDisabled(boolean disabled) {
 
-		this.hourInput.setDisabled(disabled);
-		this.minuteInput.setDisabled(disabled);
-		this.secondInput.setDisabled(disabled);
+		this.timeInput.setDisabled(disabled);
 	}
 
 	@Override
 	public Optional<Time> getValue() {
 
+		var tokens = tokenizeInputValue();
 		try {
-			Integer hours = hourInput.getValue().orElse(null);
-			Integer minutes = minuteInput.getValue().orElse(null);
-			Integer seconds = secondInput.getValue().orElse(null);
-
-			if (hours != null && minutes != null && seconds != null) {
-				return Optional.of(new Time(hours, minutes, seconds));
-			} else if (hours == null && minutes == null && seconds == null) {
+			if (tokens.isEmpty() || tokens.stream().allMatch(String::isEmpty)) {
```

Comment:
why `tokens.stream().allMatch(String::isEmpty)`? so the user can input `::` and it would be considered empty? IMO that should not be the case

Reply:
Good to know, wanted to replicate current behavior first